clients/tabby-agent/src/lsp/Server.ts

```diff
@@ -344,54 +344,97 @@ export class Server {
     }
   }
 
+  private shouldConfigurationUpdate(
+    clientProvidedConfig: ClientProvidedConfig,
+    key1: keyof ClientProvidedConfig,
+    key2?: keyof ClientProvidedConfig,
+  ): boolean {
+    const prop = clientProvidedConfig?.[key1];
+    const prevProp = this.clientProvidedConfig?.[key1];
+    return key2
+      ? prop?.[key2 as keyof typeof prop] !== prevProp?.[key2 as keyof typeof prevProp]
+      : clientProvidedConfig?.[key1] !== this.clientProvidedConfig?.[key1];
+  }
+
   private async updateConfiguration(params: DidChangeConfigurationParams) {
     const clientProvidedConfig: ClientProvidedConfig | null = params.settings;
-    if (
-      clientProvidedConfig?.server?.endpoint !== undefined &&
-      clientProvidedConfig.server.endpoint !== this.clientProvidedConfig?.server?.endpoint
-    ) {
-      if (clientProvidedConfig.server.endpoint.trim().length > 0) {
-        this.agent.updateConfig("server.endpoint", clientProvidedConfig.server.endpoint);
-      } else {
-        this.agent.clearConfig("server.endpoint");
-      }
-    }
-    if (
-      clientProvidedConfig?.server?.token !== undefined &&
-      clientProvidedConfig.server.token !== this.clientProvidedConfig?.server?.token
-    ) {
-      if (clientProvidedConfig.server.token.trim().length > 0) {
-        this.agent.updateConfig("server.token", clientProvidedConfig.server.token);
-      } else {
-        this.agent.clearConfig("server.token");
-      }
-    }
-    if (
-      clientProvidedConfig?.anonymousUsageTracking?.disable !== undefined &&
-      clientProvidedConfig.anonymousUsageTracking.disable !== this.clientProvidedConfig?.anonymousUsageTracking?.disable
-    ) {
-      if (clientProvidedConfig.anonymousUsageTracking.disable) {
-        this.agent.updateConfig("anonymousUsageTracking.disable", true);
-      } else {
-        this.agent.clearConfig("anonymousUsageTracking.disable");
+
+    if (!clientProvidedConfig) return;
+
+    const fieldsToCheck = [
+      {
+        key: "server.endpoint",
+        validation: (key: string, x: string) => {
```

Comment:
```suggestion
        validation: (key: keyof ClientProvidedConfig, x: string) => {
```

Just use explict type?

Reply:
Updated.